app/controllers/pieces_controller.rb

```diff
@@ -4,6 +4,37 @@ def show
     @pieces = @piece.game.pieces # display all the pieces on the board
   end
 
+  def update
+    @piece = Piece.find(params[:id])
+    @game = @piece.game
+    @color = @piece.color
+
+    if @color == 0
+    	@color = 'white'
+    else
+    	@color = 'black'
+    end
+
+    pos_x = params[:pos_x].to_i
+    pos_y = params[:pos_y].to_i
+
+    Piece.transaction do
```

Comment:
I don't think you need to wrap this in a transaction.  Generally speaking I only wrap things in a transaction if i'm acting on two or more rows in the database via two or more distinct calls to the database.  Additionally just FYI all of the active record callbacks (:after_save, :before_create, :before_save....) automatically use transactions, not applicable for this code but something good to know....